spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/database/support/db2/Db2PagingQueryProviderTest.java

```diff
@@ -0,0 +1,139 @@
+import org.springframework.test.context.jdbc.SqlConfig;
+import org.testcontainers.containers.Db2Container;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import javax.sql.DataSource;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.context.jdbc.SqlConfig.ErrorMode.CONTINUE_ON_ERROR;
+
+/**
+ * @author Tyler Carpenter-Rivers
+ * @author Ryan DCruz
+ * @see Db2PagingQueryProvider
+ */
+@SpringBootTest(
+	classes = {TaskStartApplication.class, Db2PagingQueryProviderTest.TestConfig.class},
+	properties = "spring.cloud.task.events.enabled=false"
+)
+@Testcontainers(disabledWithoutDocker = true)
+@DirtiesContext
+@Sql(
+	scripts = "classpath:/org/springframework/cloud/task/schema-db2.sql",
+	config = @SqlConfig(errorMode = CONTINUE_ON_ERROR)
+)
+class Db2PagingQueryProviderTest {
+
+	@Container
+	public static Db2Container db2Container =
+		new Db2Container(DockerImageName.parse("ibmcom/db2:11.5.7.0"))
+			.withDatabaseName("test")
+			.withStartupTimeoutSeconds(30)
+			.withExposedPorts(50000);
+
+	@DynamicPropertySource
+	static void properties(DynamicPropertyRegistry registry) {
+		registry.add("spring.datasource.url", db2Container::getJdbcUrl);
+		registry.add("spring.datasource.username", db2Container::getUsername);
+		registry.add("spring.datasource.password", db2Container::getPassword);
+	}
+
+	@Autowired
+	private SimpleTaskExplorer simpleTaskExplorer;
+
+	@Autowired
+	private TaskExecutionDao taskExecutionDao;
+
+	@DisplayName("Scenario: The page request size is less than the total number of elements" +
+		"Given the total number of elements is 3" +
+		"And the page request size is 2" +
+		"When the query is executed" +
+		"Then the returned page has 2 elements")
+	@Test
+	void pageRequest() {
+		// setup test data
+		taskExecutionDao.createTaskExecution(
+			UUID.randomUUID().toString(), new Date(), List.of(), UUID.randomUUID().toString());
+		taskExecutionDao.createTaskExecution(
+			UUID.randomUUID().toString(), new Date(), List.of(), UUID.randomUUID().toString());
+		taskExecutionDao.createTaskExecution(
+			UUID.randomUUID().toString(), new Date(), List.of(), UUID.randomUUID().toString());
+
+		PageRequest pageRequest = PageRequest.of(0, 2);
+
+		// run subject under test
+		Page<TaskExecution> page = simpleTaskExplorer.findAll(pageRequest);
+
+		// assert and verify
+		assertThat(page.getTotalElements())
+			.as("Expected the total number of elements to exceed the page request size")
+			.isGreaterThan(pageRequest.getPageSize());
+
+		assertThat(pageRequest.getPageSize())
+			.as("Expected the size of the returned page to equal what was requested")
+			.isEqualTo(page.getNumberOfElements());
+	}
+
```

Comment:
May want to add an test that checks for a result set less than the requested page count.   For example:
```
	@Test
	void pageRequestSinglePageEntries() {
		// setup test data
		PageRequest pageRequest = PageRequest.of(0, 2);

		// run subject under test
		Page<TaskExecution> page = simpleTaskExplorer.findAll(pageRequest);

		// assert and verify

		assertThat(page.getTotalElements())
			.as("Should have one element returned")
			.isEqualTo(1);

		assertThat(page.getTotalPages())
			.as("Should return only one page")
			.isEqualTo(1);
	}
```

Reply:
Added this test, as well as a script in an `@Sql` that cleans up the tables between each test.